### PR TITLE
explicitly require Python 3.6 and drop six dependency

### PR DESCRIPTION
**Pull request by StrikerRUS**
Linking 
https://github.com/scikit-learn-contrib/lightning/pull/142#issue-417685844
> This PR should be treated as a hotfix, and ideally `lightning` should drop the support of Python 2 with `six` dependency.

and
https://github.com/scikit-learn-contrib/lightning/pull/161#issuecomment-834138269
> Libraries have agonized over this for a long time before py2 was sunset. Now that it has, I think it's ok and better to officially drop support.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -66,8 +66,8 @@ penalty on the News20 dataset (c.f., `Blondel et al. 2013
 Dependencies
 ------------
 
-lightning requires Python >= 2.7, setuptools, Numpy >= 1.3, SciPy >= 0.7 and
-scikit-learn >= 0.15. Building from source also requires Cython and a working C/C++ compiler. To run the tests you will also need pytest.
+lightning requires Python >= 3.6, setuptools, Numpy >= 1.12, SciPy >= 0.19 and
+scikit-learn >= 0.19. Building from source also requires Cython and a working C/C++ compiler. To run the tests you will also need pytest.
```
Comment (StrikerRUS) [inline README.rst:70]: These are the first versions for which Python 3.6 wheels were uploaded to PyPI.

**Comment by StrikerRUS** [comment]
Looks like http://contrib.scikit-learn.org/lightning/ is not in sync with the current `master`, e.g.

image


image


Are docs builds there triggered manually or is there any sphinx compiling error? I'm not able to see the logs (or don't know where to search for them).

**Comment by StrikerRUS** [comment]
@mblondel @vene
Thanks for your time and consideration!

**Comment by mblondel** [comment]
The documentation needs to be manually updated in the `gh-pages` branch.

**Comment by mblondel** [comment]
Merged! I don't know how to automatically upload the website.

**Pull request merged**